cpp/src/cylon/arrow/arrow_kernels.cpp

```diff
@@ -167,8 +257,12 @@ class BinaryArraySplitKernel : public ArrowArraySplitKernel {
       const int64_t arr_len = array->length();
       for (int64_t i = 0; i < arr_len; i++, offset++) {
         ARROW_OFFSET_T length = 0;
-        const uint8_t *value = casted_array->GetValue(i, &length);
-        RETURN_CYLON_STATUS_IF_ARROW_FAILED(builders[target_partitions[offset]]->Append(value, length));
+        if (nulls_present && casted_array->IsNull(i)) {
+          RETURN_CYLON_STATUS_IF_ARROW_FAILED(builders[target_partitions[offset]]->AppendNull());
+        } else {
+          const uint8_t *value = casted_array->GetValue(i, &length);
+          RETURN_CYLON_STATUS_IF_ARROW_FAILED(builders[target_partitions[offset]]->Append(value, length));
+        }
```

Comment:
shall we have 2 separate for loops for `nulls_present` and not. `IsNull(i)` within a hot loop could be expensive?

Reply:
well, looking back, branch prediction may figure things out and predict the branch. But I think it would be safe to have 2 separate loops for this